File: client/src/types/comment.ts
import type { VoteType } from './vote';

export interface CommentAuthor {
  id: string;
  name: string;
  mentionHandle: string | null;
}

export interface Comment {
  id: number;
  storyId: number;
  author: CommentAuthor;
  body: string;
  parentCommentId: number | null;
  createdAt: string;
  editedAt: string | null;
  replies: Comment[];
  upvoteCount: number;
  downvoteCount: number;
  currentUserVote: VoteType | null;
}

export interface PostCommentRequest {
  storyId: number;
  body: string;
  parentCommentId?: number;
}

File: src/UpToU.Core/Entities/Comment.cs
namespace UpToU.Core.Entities;

public class Comment
{
    public int Id { get; set; }
    public int StoryId { get; set; }
    public Story Story { get; set; } = null!;
    public string AuthorId { get; set; } = string.Empty;
    public ApplicationUser Author { get; set; } = null!;
    public int? ParentCommentId { get; set; }
    public Comment? ParentComment { get; set; }
    public ICollection<Comment> Replies { get; set; } = new List<Comment>();
    public string Body { get; set; } = string.Empty;
    public bool IsDeleted { get; set; } = false;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? EditedAt { get; set; }
    public ICollection<CommentVote> CommentVotes { get; set; } = new List<CommentVote>();
}
